src/OrbitBase/TypedefTest.cpp

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) 2022 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include <gtest/gtest.h>
+
+#include <utility>
+#include <vector>
+
+#include "OrbitBase/Typedef.h"
+
+namespace {
+struct TestTag {};
+
+struct AddStruct {
+  int operator()(int i, int j) { return i + j; }
+
+  static int Add(int i, const int& j) { return i + j; }
+};
+
+}  // namespace
+
+static int Add(int i, int j) { return i + j; }
+
+namespace orbit_base {
+template <typename T>
+using Wrapper = Typedef<TestTag, T>;
+
+TEST(TypedefTest, CanInstantiate) {
+  const int kConstInt = 1;
+  Wrapper<int> wrapper_of_const(kConstInt);
+  EXPECT_EQ(*wrapper_of_const, kConstInt);
+
+  constexpr int kConstexprInt = 1;
+  Wrapper<int> wrapper_of_constexpr(kConstexprInt);
+  EXPECT_EQ(*wrapper_of_constexpr, kConstexprInt);
+
+  int non_const = 1;
+  Wrapper<int> wrapper_of_non_const(non_const);
+  EXPECT_EQ(*wrapper_of_non_const, non_const);
+
+  Wrapper<int> wrapper_of_literal(1);
+  EXPECT_EQ(*wrapper_of_literal, 1);
+
+  Wrapper<std::string> wrapper_of_string_literal("foo");
+  EXPECT_EQ(*wrapper_of_string_literal, "foo");
+}
+
+TEST(TypedefTest, CallIsCorrect) {
+  const int kFirst = 1;
+  const int kSecond = 2;
+  const int kSum = kFirst + kSecond;
+
+  const Wrapper<int> kFirstWrapped(kFirst);
+  const Wrapper<int> kSecondWrapped(kSecond);
+
+  {
+    auto add = [](int i, int j) { return i + j; };
+    const Wrapper<int> sum_wrapped = Call(add, kFirstWrapped, kSecondWrapped);
+    EXPECT_EQ(*sum_wrapped, kSum);
+  }
+
+  {
+    auto add = [](const int& i, const int& j) { return i + j; };
+    const Wrapper<int> sum_wrapped = Call(add, kFirstWrapped, kSecondWrapped);
+    EXPECT_EQ(*sum_wrapped, kSum);
+  }
+
+  {
+    const Wrapper<int> sum_wrapped = Call(Add, kFirstWrapped, kSecondWrapped);
+    EXPECT_EQ(*sum_wrapped, kSum);
+  }
+
+  {
+    const Wrapper<int> sum_wrapped = Call(AddStruct::Add, kFirstWrapped, kSecondWrapped);
```

Comment:
Not sure what `AddStruct` is about. You're calling a static function which should not behave differently to a  free function. Do you wanted to test a member function call? Then you need to do something like this:

```cpp
struct Integer {
  int value;
  
  Integer Add(const Integer& other) const { return Integer{value + other.value}; }
};
```

Reply:
> calling a static function which should not behave differently to a free function
NVM then. I'll remove it.

Reply:
Done.